test/Kestrel.Core.Tests/Http1ConnectionTests.cs

```diff
@@ -81,6 +81,37 @@ public void Dispose()
             _pipelineFactory.Dispose();
         }
 
+        [Fact]
+        public async Task TakeMessageHeadersSucceedsWhenHeaderValueContainsUTF8()
```

Comment:
Adding these tests to Http1ConnectionTests even though they are mostly testing HttpProtocol since this seems like the most convenient place.

Reply:
I've added this test in my PR https://github.com/aspnet/KestrelHttpServer/pull/2870